### ISSUE TITLE
Use forked WebDriverIO or newer version of WinAppDriver for E2E testing

### ISSUE THREAD

**Issue by licanhua** [microsoft/react-native-windows#3019]
[PR 4369](https://github.com/webdriverio/webdriverio/pull/4369) described the problem between WinAppDriver and WebDriverIO.
I always created PR on WinAppDriver to fix the problem, but still no ETA to release it.
Possible adopt the newest WinAppDriver or use forked WebDriverIO.

**Comment by christian-bromann** [comment]
@licanhua I don't mind adding a workaround for WinAppDriver for WebdriverIO but this workaround should be an opt-in solution so that the default behavior remains the same.

**Comment by licanhua** [comment]
Before WinAppDriver fix it, I worked around the problem in #2975 by
```
    "webdriverio": "^5.10.1",
    "webdriver": "git+https://github.com/react-native-windows/webdriver.git",
```
I keep this PR open until I adopted the new WinAppDriver.

**Comment by licanhua** [comment]
WinAppDriver and its marketplace script are not ready, postpone it to next iteration

**Comment by licanhua** [comment]
Need to re-enable webdriverio bump which is disabled in #3386

**Comment by licanhua** [comment]
WinAppDriver fixed it on [v1.2-RC](https://github.com/microsoft/WinAppDriver/releases)
appium-windows-driver fixed in [1.7.0](https://github.com/appium/appium-windows-driver/pull/60) .

Because WinAppDriver requires admin permission to install it, and our build machine is still using WinAppDriver 1.1. 
This task can be unblocked if any of below tasked can be resolved:
1. WinAppDriver can be deployed without admin permission. https://github.com/microsoft/WinAppDriver/issues/913
2. [Microsoft hosted agent](https://github.com/Microsoft/azure-pipelines-image-generation/blob/master/images/win/Vs2019-Server2019-Readme.md) is using WinAppDriver 1.2-RC. 
3. #3759, we host our own build machine and we can install WinAppDriver on it.

**Comment by NickGerleman** [comment]
I just added some selective dependency resolutions to our main package.json to ensure we use old versions of some dependencies if we regen our lockfile. We should make sure to remove this when we upgrade.

Confirmed the hosted agent is still on WinAppDriver 1.1

**Comment by NickGerleman** [comment]
We have new WinAppDriver now.
